examples/controlnet/add_controlnet.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+
+import argparse
+import os
+
+from diffusers import ControlNetModel, StableDiffusionControlNetPipeline, UNet2DConditionModel
+
+
+def parse_args(input_args=None):
+    parser = argparse.ArgumentParser(description="Add controlnet to existing model by copying unet weights.")
+    parser.add_argument(
+        "--pretrained_model_name_or_path",
+        type=str,
+        default=None,
+        required=True,
+        help="Path to pretrained model or model identifier from huggingface.co/models.",
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=str,
+        default=None,
+        required=True,
+        help="The output directory where the model will be written.",
+    )
+    parser.add_argument(
+        "--revision",
+        type=str,
+        default=None,
+        required=False,
+        help=(
+            "Revision of pretrained model identifier from huggingface.co/models. Trainable model components should be"
+            " float32 precision."
+        ),
+    )
+    parser.add_argument(
+        "--controlnet_only",
+        action="store_true",
+        help=("Save only the controlnet weights."),
+    )
+    if input_args is not None:
+        args = parser.parse_args(input_args)
+    else:
+        args = parser.parse_args()
+
+    return args
+
+
+def main(args):
+    device_map = "cpu"
+
+    try:
+        config, unused_kwargs = ControlNetModel.load_config(
```

Comment:
Nice - could we maybe integrate this code directly into the training script? Think for training by default we should start from a copy of the original unet.